src/seedu/addressbook/Main.java

```diff
@@ -76,11 +77,13 @@ private void exit() {
         System.exit(0);
     }
 
-    /** Reads the user command and executes it, until the user issues the exit command.  */
-    private void runCommandLoopUntilExitCommand() {
+    /** Reads the user command and executes it, until the user issues the exit command.  
+     * @throws FileDeletedException */
+    private void runCommandLoopUntilExitCommand() throws FileDeletedException {
         Command command;
         do {
             String userCommandText = ui.getUserCommand();
+        	storage.checkFileDeleted();
             command = new Parser().parseCommand(userCommandText);
```

Comment:
Is there still a need for creating an instance of Parser, if the methods in it are already `static`?